src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -218,16 +212,16 @@ public boolean equals(Object obj) {
                Objects.equals(queryName, other.queryName);
     }
 
-    /** Read the given parameters. */
     @Override
-    public void readFrom(StreamInput in) throws IOException {
-        fieldName = in.readString();
-        value = in.readGenericValue();
-        boost = in.readFloat();
-        queryName = in.readOptionalString();
+    public QB readFrom(StreamInput in) throws IOException {
+        QB emptyBuilder = createBuilder(in.readString(), in.readGenericValue());
+        emptyBuilder.boost(in.readFloat());
```

Comment:
I wonder if we should use the setters in deserialization if we write the fields directly in the serialization. This works, but we already had cases with the internal conversion of BytesRef in the Terms/RangeQuery where this might be trappy. Otherwise, should be also use getters in writeTo()?